src/main/java/hudson/plugins/mercurial/MercurialSCM.java

```diff
@@ -850,12 +853,29 @@ private void clone(Run<?, ?> build, Launcher launcher, FilePath repository, Node
         if (cachedSource != null && !cachedSource.isUseSharing()) {
             FilePath hgrc = repository.child(".hg/hgrc");
             if (hgrc.exists()) {
-                String hgrcText = hgrc.readToString();
-                if (!hgrcText.contains(cachedSource.getRepoLocation())) {
-                    listener.error(".hg/hgrc did not contain " + cachedSource.getRepoLocation() + " as expected:\n" + hgrcText);
-                    throw new AbortException(".hg/hgrc did not contain " + cachedSource.getRepoLocation() + " as expected:\n" + hgrcText);
+                StringBuffer hgrcBuffer = new StringBuffer();
+                Boolean defaultPathFound = Boolean.FALSE;
+                try ( InputStreamReader input = new InputStreamReader(hgrc.read(), StandardCharsets.UTF_8);
+                      BufferedReader reader = new BufferedReader(input)
```

Comment:
If you are going to parse the file you may as well use ini4j, as elsewhere in the plugin. Easier and more robust.